openupgrade_scripts/scripts/iap/18.0.1.1/pre-migration.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2025 ForgeFlow S.L. (https://www.forgeflow.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE iap_account
+        SET name = service_name
+        WHERE COALESCE(name, '') != ''
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        DELETE FROM iap_account
+        WHERE service_name = 'reveal'
+        """,
+    )
```

Comment:
instead of deleting those accounts, you should rather exclude `reveal` in the post migration query and link accounts with this service name to the service created in xml data